New York Times Article Page Clone

ElijahTobs opened the pull request:
**NYT Clone**

- add navbar
- add header
- add sidebar
- add footer
- use float, flex and grid

ebeagusamuel committed:
add html structure for the navbar and some styling

ElijahTobs committed:
Add article tag

- add icons below the first image
- add some paragraphs within the article tag

ebeagusamuel committed:
add more html markup and css styling

-an aside tag was added which holds the content that floats to the right in the page.
-css float styling was used to get it to the right side of the page.
-images and videos was also embedded into the page.
-css grid and flexbox was used to arrange the images and videos into the desired layout.

ebeagusamuel committed:
add more html markup and css styling

-used more css grid styling to build a layout where images and text were arranged.

ElijahTobs committed:
Add the footer

ElijahTobs committed:
Update README.md

christianotieno reviewed (commented):
## Pull Request under review by Christian Otieno (TSE)

christianotieno reviewed (changes requested):
# **CHANGES REQUIRED** :warning:
Hey @ElijahTobs  & @ebeagusamuel  :wave:
-
You have done a splendid job! :tada: Keep it up.
We, however, do have some few changes to carry out before we proceed (see review comment).
##
Let's improve on the already awesome implementation of the navbar:
From this:
![header_clone](https://user-images.githubusercontent.com/22345961/78457837-d866df00-76b5-11ea-9ab3-b634cef12aa2.png)
A bit of tweaking to make it appear more like this:
![header](https://user-images.githubusercontent.com/22345961/78457842-e7e62800-76b5-11ea-8d9c-5efb8c021c4a.png)

### NOTES: I am conducting this review with a (1920 x 1080) screen display resolution. So bear with me if your resolution might differ with mines. You are free to reach out to me here or on slack for further discussion on the same.
The big main centred image and text appears not to be properly centred. 
- This is how it is appearing on my screen:
![image_clone](https://user-images.githubusercontent.com/22345961/78457755-38a95100-76b5-11ea-9585-a5aa79d4e5e3.png)

- This is how the original appears on my screen:
![image](https://user-images.githubusercontent.com/22345961/78457776-57a7e300-76b5-11ea-83ee-3a620cb4c0ad.png)

The same also applies to the editor's pick section:
Yours:
![editor_clone](https://user-images.githubusercontent.com/22345961/78457882-2ed41d80-76b6-11ea-841a-6534c3cb5f3f.png)

Original:
![editor](https://user-images.githubusercontent.com/22345961/78457886-385d8580-76b6-11ea-86de-359308e5f8ef.png)

- Next in line, we have the 'More in Space and Astronomy' section:
Yours:
![more_space_clone](https://user-images.githubusercontent.com/22345961/78457916-680c8d80-76b6-11ea-9eb9-df8839165627.png)

Original:
![more_space](https://user-images.githubusercontent.com/22345961/78457918-6fcc3200-76b6-11ea-8ff3-c5a4e67514f7.png)

- Lastly, we have the footer section:
- Yours:
![footer_clone](https://user-images.githubusercontent.com/22345961/78457943-95593b80-76b6-11ea-9efd-6b93205fce23.png)

- Original:
![footer](https://user-images.githubusercontent.com/22345961/78457950-a013d080-76b6-11ea-8f16-339d73d541de.png)

I think the cause of this would be how we implemented the responsiveness in this project. The page appears to not also be responsive in medium and small screens.  

## [REQUIRED] - VALIDATE -The markup.
The markup currently has some errors and warning that needs getting sorted out.
- Kindly let us see to it:
![errors](https://user-images.githubusercontent.com/22345961/78457989-f2ed8800-76b6-11ea-8f07-1c3facc886f3.png)

## [OPTIONAL] - REDUCE - The use of comments in your code
- Normally comments are a good thing, they help us remember bits of our code when we come back to it. However, I'd recommend we minimise overusing comments as it makes the codebase ugly. 
It’s better to delete the majority of your comments. If you need to review a previous version of your code, git history is perfect for this, thank you,

## 
Please make these changes and submit another review request. Thank you!
- If you have any questions, let me know here or on slack and I will be glad to clarify with you. You can also reach me and connect with me on my social media or email. :eyes:
##
Happy coding :smile:
**[Christian Otieno](https://github.com/ChristianOtieno)**
[Twitter](https://twitter.com/iamchrisotieno) [linkedin](https://www.linkedin.com/in/christianotieno/) [GitLab](https://gitlab.com/christianotieno)
---
*P.S. all requested changes not marked by [OPTIONAL] or are not positive comments are **REQUIRED***

ElijahTobs committed:
implement required change

ElijahTobs commented:
I have implemented all required and optional changes

ebeagusamuel committed:
add screenshot of the cloned page

ebeagusamuel committed:
add screenshot of the cloned page

ebeagusamuel committed:
add screenshot of the cloned page

sumancrest0001 commented:
STATUS: CHANGES REQUIRED
Great job and those improvements already made your project more beautiful. :clap: :clap: 
But still, we need to make improvements to get :+1:  for the project.

Kindly make the following changes.

[REQUIRED] Instead of the account link we have to add **log in button** and nav bar must be sticky at the top. 
**Please have a look :point_down: :point_down:** 

![Screenshot from 2020-04-05 19-56-18](https://user-images.githubusercontent.com/37747619/78500683-b0d84b00-7777-11ea-8ff6-405fbfce5d3b.png)

**In original project :point_down:** 
![Screenshot from 2020-04-05 19-56-58](https://user-images.githubusercontent.com/37747619/78500710-ccdbec80-7777-11ea-826a-a1af0ce29075.png)

sumancrest0001 commented on index.html:
```diff
@@ -0,0 +1,540 @@
+          <i class="fas fa-search search"></i>
+          <span>SPACE & COSMOS</span>
+        </span>
+        <span class="center-nav-bar">
+          <img src="./img/nytLogo.PNG" alt="nytLogo">
+        </span>
```
[REQUIRED] The logo image should be a link. Not just only an image.

sumancrest0001 commented:
[REQUIRED] We have to add a sticky footer section at the bottom.

In original project :point_down:  :point_down: 
![Screenshot from 2020-04-05 20-01-25](https://user-images.githubusercontent.com/37747619/78500797-4bd12500-7778-11ea-9ff3-40d279f09edf.png)

[link to original page](https://www.nytimes.com/2014/03/18/science/space/detection-of-waves-in-space-buttresses-landmark-theory-of-big-bang.html)

sumancrest0001 commented:
**Please make above changes and submit for another review.**

Need help? message me on slack **Suman Shrestha**

ebeagusamuel committed:
modication made to the navbar
- I added a sign-in button to the navbar.
- I made the nav-bar fixed to te top of the page.
- I made the the NYT logo in the navbar to be clickable.

ElijahTobs committed:
add sticky footer

ElijahTobs committed:
add sticky footer

ebeagusamuel committed:
delete code comments that are not useful

uryelah commented:
Hello **Samuel Ebeagu** & **Elijah Ayandokun**, I'm Sarah, your last reviewer for this PR 👩‍🚀  

**Good job with the fixes from the last reviews! 🌟**

The project is looking great, I love how you also added the hover interactions and paid attention to font size and style.

------
## Your project is approved! 🎊 🥅 💛 
### 
**You may close this Pull Request and merge the changes**

----

### Optional Changes

- [ ] The navbar and fixed footer should both take 100% of the screen width. Sometimes you can miss that according to your screen size. A fast trick to see how your page looks in a slightly wider screen is clicking `ctrl+`, or you could use your browser responsiveness tools.

Notice how there's space in the sides and you can even see the box-shadow 'leaking':
![Screenshot_2020-04-06 NYTClone](https://user-images.githubusercontent.com/16193212/78523786-44c20b00-77a8-11ea-8b50-5022c0c86a68.png)
![Screenshot_2020-04-06 NYTClone(2)](https://user-images.githubusercontent.com/16193212/78523788-44c20b00-77a8-11ea-9a31-f2c79ee77bd5.png)

- [ ] Make the icons with the circular border bellow the article all have a circle around then by setting a fixed width and height. Notice how the Facebook icon bellow has an ellipse instead:
![Screenshot_2020-04-06 NYTClone(1)](https://user-images.githubusercontent.com/16193212/78523785-44297480-77a8-11ea-951c-1513b327e1eb.png)
----

**Keep being awesome devs and good luck with your next project!**

Cheers and Happy coding!👏👏👏

Code review by **Sarah Uryelah Silva Chamorro**

(Feel free to contact me on slack by the handle of  **sarah** if you have any additional question)
